### PR TITLE
feat(license): add Apache 2.0 license and reuse check

### PR DESCRIPTION
**Pull request by rgallor**

### CODE REVIEW

**Comment by harlem88** [comment]
Add [REUSE Compliance Check](https://github.com/fsfe/reuse-action#reuse-compliance-check) to github action, see here [https://github.com/astarte-platform/astarte-message-hub-proto/blob/master/.github/workflows/reuse-lint.yaml](https://github.com/astarte-platform/astarte-message-hub-proto/blob/master/.github/workflows/reuse-lint.yaml)

File: .github/workflows/reuse-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: REUSE Compliance Check
+        uses: fsfe/reuse-action@v2
```
Comment (harlem88) [inline .github/workflows/reuse-lint.yaml:26]: Add new line at end of file
```suggestion
        uses: fsfe/reuse-action@v2

```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: weekly
```
Comment (harlem88) [inline .github/dependabot.yaml:9]: Add new line at end of file
```suggestion
      interval: weekly
      
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .github/dependabot.yaml (hunk 2 above)
Comment (rgallor) [reply .github/dependabot.yaml]: Done

File: LICENSES/Apache-2.0.txt [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (harlem88) [inline LICENSES/Apache-2.0.txt:1]: I am not sure but i think you added too many licenses. The licenses folder should be use only if you have different licenses in the repo for example Apache and LGPL etc..

Comment (noaccOS) [reply LICENSES/Apache-2.0.txt:1]: The LICENCES directory was generated by `reuse init`. Unfortunately `reuse lint` only works with the files under LICENSES and ignores the top level LICENSE file

**Review by harlem88 — APPROVED** [review]

**Review by noaccOS — APPROVED** [review]

**Pull request merged**